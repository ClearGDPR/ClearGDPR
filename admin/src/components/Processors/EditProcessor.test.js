import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditProcessorForm, { EditProcessor } from './EditProcessor';

const setup = propOverrides => {
  const props = Object.assign({}, propOverrides);
  const component = shallow(<EditProcessor {...props} />);

  return { props, component, mount };
};

const setupForm = propOverrides => {
  let formApi;
  const props = Object.assign({}, propOverrides);
  const component = mount(
    <EditProcessorForm
      {...props}
      getApi={api => {
        formApi = api;
      }}
    />
  );

  return { formApi, props, component };
};


describe('(Component) Edit Processor Form', () => {
  it('should render correctly when no props provided', async () => {
    const { component } = setup();
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to false', async () => {
    const { component } = setup({ isLoading: false });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to true', async () => {
    const { component } = setup({ isLoading: true });
    expect(toJson(component)).toMatchSnapshot();
  });
});
