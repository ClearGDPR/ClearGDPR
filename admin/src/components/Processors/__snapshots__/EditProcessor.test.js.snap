// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Edit Processor Form should render correctly when no props provided 1`] = `
<React.Fragment>
  <TextInput
    field="name"
    label="Processor name"
    placeholder="Processor name"
    required={true}
    type="text"
  />
  <label
    htmlFor="description"
  >
    Description:
  </label>
  <TextArea
    field="description"
    id="description"
  />
  <TextInput
    field="logoUrl"
    label="Logo URL"
    placeholder="Processor Logo (URL)"
    required={true}
    type="url"
  />
  <label>
    Scopes:
  </label>
  <[object Object]
    scope="scopes"
  >
    <div
      key="0"
    >
      <Checkbox
        field="user:fullName"
        id="scopes-user:fullName"
      />
      <label
        htmlFor="scopes-user:fullName"
        key="0"
      >
        user:fullName
      </label>
    </div>
    <div
      key="1"
    >
      <Checkbox
        field="user:email"
        id="scopes-user:email"
      />
      <label
        htmlFor="scopes-user:email"
        key="1"
      >
        user:email
      </label>
    </div>
    <div
      key="2"
    >
      <Checkbox
        field="user:phoneNumber"
        id="scopes-user:phoneNumber"
      />
      <label
        htmlFor="scopes-user:phoneNumber"
        key="2"
      >
        user:phoneNumber
      </label>
    </div>
  </[object Object]>
  <button
    className="btn ui-action"
    type="submit"
  >
    Submit
  </button>
</React.Fragment>
`;

exports[`(Component) Edit Processor Form should render correctly when the prop isLoading is set to false 1`] = `
<React.Fragment>
  <TextInput
    field="name"
    label="Processor name"
    placeholder="Processor name"
    required={true}
    type="text"
  />
  <label
    htmlFor="description"
  >
    Description:
  </label>
  <TextArea
    field="description"
    id="description"
  />
  <TextInput
    field="logoUrl"
    label="Logo URL"
    placeholder="Processor Logo (URL)"
    required={true}
    type="url"
  />
  <label>
    Scopes:
  </label>
  <[object Object]
    scope="scopes"
  >
    <div
      key="0"
    >
      <Checkbox
        field="user:fullName"
        id="scopes-user:fullName"
      />
      <label
        htmlFor="scopes-user:fullName"
        key="0"
      >
        user:fullName
      </label>
    </div>
    <div
      key="1"
    >
      <Checkbox
        field="user:email"
        id="scopes-user:email"
      />
      <label
        htmlFor="scopes-user:email"
        key="1"
      >
        user:email
      </label>
    </div>
    <div
      key="2"
    >
      <Checkbox
        field="user:phoneNumber"
        id="scopes-user:phoneNumber"
      />
      <label
        htmlFor="scopes-user:phoneNumber"
        key="2"
      >
        user:phoneNumber
      </label>
    </div>
  </[object Object]>
  <button
    className="btn ui-action"
    type="submit"
  >
    Submit
  </button>
</React.Fragment>
`;

exports[`(Component) Edit Processor Form should render correctly when the prop isLoading is set to true 1`] = `
<React.Fragment>
  <TextInput
    field="name"
    label="Processor name"
    placeholder="Processor name"
    required={true}
    type="text"
  />
  <label
    htmlFor="description"
  >
    Description:
  </label>
  <TextArea
    field="description"
    id="description"
  />
  <TextInput
    field="logoUrl"
    label="Logo URL"
    placeholder="Processor Logo (URL)"
    required={true}
    type="url"
  />
  <label>
    Scopes:
  </label>
  <[object Object]
    scope="scopes"
  >
    <div
      key="0"
    >
      <Checkbox
        field="user:fullName"
        id="scopes-user:fullName"
      />
      <label
        htmlFor="scopes-user:fullName"
        key="0"
      >
        user:fullName
      </label>
    </div>
    <div
      key="1"
    >
      <Checkbox
        field="user:email"
        id="scopes-user:email"
      />
      <label
        htmlFor="scopes-user:email"
        key="1"
      >
        user:email
      </label>
    </div>
    <div
      key="2"
    >
      <Checkbox
        field="user:phoneNumber"
        id="scopes-user:phoneNumber"
      />
      <label
        htmlFor="scopes-user:phoneNumber"
        key="2"
      >
        user:phoneNumber
      </label>
    </div>
  </[object Object]>
  <button
    className="btn ui-action"
    type="submit"
  >
    Submit
  </button>
</React.Fragment>
`;
